tests/unit/Core/Checkout/Promotion/Cart/PromotionProcessorTest.php

```diff
@@ -126,4 +127,70 @@ public function testProcessWithCartZeroPriceAndPromotionIsNotGlobal(): void
         static::assertCount(1, $toCalculateCart->getErrors());
         static::assertInstanceOf(PromotionsOnCartPriceZeroError::class, $toCalculateCart->getErrors()->first());
     }
+
+    /**
+     * Test that promotions with considerAdvancedRules=true but uninitialized filter keys
+     * work correctly when adding items to cart without throwing initialization errors.
+     *
+     * This test simulates the real cart scenario where your targeted fix prevents crashes.
+     */
+    public function testProcessWithConsiderAdvancedRulesButUninitializedFilterKeys(): void
```

Comment:
Write an integration test instead, as it should test the calculation though all services. This test is kinda pointless as you're mocking the essential parts of the promotion process :thinking: